### DISCUSSION TITLE
Unable to load spatialite extension into a geopackage

### DISCUSSION THREAD

**Discussion by ryanjshields** [pawelsalawa/sqlitestudio#4727] [General]
### Details
I'm connected to a geopackage and I want to load the spatialite extension but get the error: 'Cannot execute query on undefined or invalid database. I've tried using the full file path to the .dll file in load_extension() as well as adding it to the PATH. Both produce the same error.

Note: I reran it before submitting and the error has now changed to: Error while executing SQL query on database 'BlueStar': The specified module could not be found. The only difference is expanded the gpkg and looked through some of the tables.

### Steps to reproduce
download and unzip geopackage [BlueStar.zip](https://github.com/pawelsalawa/sqlitestudio/files/11004042/BlueStar.zip)
download mod_spatialite.dll from [here](http://www.gaia-gis.it/gaia-sins/windows-bin-amd64/)
Connect to the BlueStar.gpkg
Run SELECT load_extension('mod_spatialite.dll');  (try the full file path or add to the environment variable)

### Operating system
Edition	Windows 10 Home
Version	21H2
Installed on	‎8/‎12/‎2021
OS build	19044.2728
Experience	Windows Feature Experience Pack 120.2212.4190.0

### SQLiteStudio version
3.4.3

**Comment by pawelsalawa** [comment]
The mod_spatialite uses different versions of certain dlls, causing incompatibility during attempt to load it.
Solution is to make a copy of your SQLiteStudio folder and then copy all files from mod_spatialite package into this new copy folder of SQLiteStudio, overwritting all files there. Then run SQLiteStudio from that folder and load mod_spatialite.dll and it should work. I recommend you to use extension manager or that:
![image](https://user-images.githubusercontent.com/481846/226097758-9b5ee941-0de5-424d-bb51-18762a8cf168.png)

**Reply by IPv777** [reply]
Thanks ! I had this problem too !
